pkg/controllers/etcd.go

```diff
@@ -103,18 +103,26 @@ func (s *EtcdService) Run(ctx context.Context, ready chan<- struct{}, stopped ch
 		return fmt.Errorf("%s failed to start: %v", s.Name(), err)
 	}
 
-	// Make sure microshift-etcd is terminated whether it failed to properly start or MicroShift stops
+	// Handle microshift-etcd termination before microshift process exits
+	go func() {
+		if err := cmd.Wait(); err != nil {
+			klog.Warningf("%v failed waiting on process to finish: %+v", s.Name(), err)
+		}
+		klog.Infof("%v process quit: %v", s.Name(), cmd.ProcessState.String())
+
+		// Exit microshift to trigger microshift-etcd restart
+		klog.Warning("microshift-etcd process terminated prematurely, restarting MicroShift")
+		os.Exit(0)
```

Comment:
Instead of abruptly exiting we should cancel the run context (https://github.com/openshift/microshift/blob/main/pkg/cmd/run.go#L118) to make sure that other parts of the process have a chance to clean up after themselves before the main() function exits. I don't know if the ctx argument to this function can do that or if we need to pass that cancel function in separately.

Reply:
We have a few other places where we call exit. And, we have a task to replace those.
Should we do it in another PR so that all exits are consistent?